launch/launch_test.go

```diff
@@ -0,0 +1,104 @@
+package launch_test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+
+	"github.com/buildpacks/lifecycle/launch"
+	h "github.com/buildpacks/lifecycle/testhelpers"
+)
+
+func TestLaunch(t *testing.T) {
+	spec.Run(t, "Launch", testLaunch, spec.Sequential(), spec.Report(report.Terminal{}))
+}
+
+func testLaunch(t *testing.T, when spec.G, it spec.S) {
+	when("Process", func() {
+		when("MarshalText", func() {
```

Comment:
I wondered that as well at first. The exporter tests already check the metadata toml in the image label `io.buildpacks.build.metadata` and the tests pass. Is there somewhere else you think we should double check?

Reply:
I saw that too... I guess we're in the clear for now but this seems like something that could bite us unexpectedly in the future. Is it possible to switch this back to `MarshalTOML`, and handle the newlines somehow?

Reply:
Hmm. I don't see a straight forward way to do this. The encoding change we need is [here](https://github.com/BurntSushi/toml/blob/360c9e349664b57aac916a5c41aac8702b179c09/encode.go#L319-L321). But it doesn't seem to be exposed via any public members.

The `MarshalTOML` interface usage seems to be intended for a single custom "primitive" type with a single value. I think our usage here is a bit odd as we turn a single `launch.Process` into an exploded toml/json.

If we wanted to, I think we could try something like

```
#old
Command          []string       `toml:"-" json:"-"` // ignored
Args             []string `toml:"args" json:"args"`

#new
Command          launch.CommandValue       `toml:"command" json:"command"`
Args             launch.ArgValue `toml:"args" json:"args"`
```

`launch.CommandValue` and `launch.ArgValue` would be the type we implement `MarshalText` on. This feels like a more supported and expected path, but right now we do this translation bit in the marshaller methods. The `launch.ArgValue` wouldn't have access to the `launch.Command` value in order to grab the extra command args.

```
		Command:          p.Command[0],
		Args:             append(p.Command[1:], p.Args[0:]...),
```

I could still give this a shot and see what I can come up with if you think it is worth the effort. Let me know!

Reply:
I think for now because we have a `MarshalJSON` method on `Process` we don't have to worry that `MarshalText` will ever be called when serializing JSON because `MarshalJSON` has precedence :) So we can call this done 🎉